sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -114,8 +114,32 @@ case class ConcatWs(children: Seq[Expression])
         boolean ${ev.isNull} = ${ev.primitive} == null;
       """
     } else {
-      // Contains a mix of strings and array<string>s. Fall back to interpreted mode for now.
-      super.genCode(ctx, ev)
+      val list = ctx.freshName("list")
+      val array = ctx.freshName("array")
+      val sep = children.head.gen(ctx)
+      val argsCode = children.tail.map(x => {
+        val gen = x.gen(ctx)
+        if (x.dataType == StringType) {
+          s"""
+            ${gen.code}
+            $list.add(${gen.primitive});
+            """
+        } else {
+          s"""
+            ${gen.code}
+            $list.addAll(scala.collection.JavaConversions.asJavaCollection(${gen.primitive}));
```

Comment:
actually i think this breaks because now we refactored the codebase to use ArrayData as the internal representation for Seq.